textscreen/txt_fileselect.c

```diff
@@ -463,12 +463,20 @@ int TXT_CanSelectFiles(void)
     return 1;
 }
 
-char *TXT_SelectFile(char *window_title, char **extensions)
+char *TXT_SelectFile(const char *window_title, char **extensions)
 {
     char *argv[4];
     char *result, *applescript;
 
-    applescript = GenerateAppleScript(window_title, extensions);
+    // We need to create a temporary copy of window_title because GenerateApple-
+    // Script works with non-const string.
+    char *window_title_copy = strdup(window_title);
```

Comment:
This is not the correct fix. You should instead make the appropriate parameters of GenerateAppleScript, GenerateSelector and EscapedString const so that constness is properly preserverd throughout.

Reply:
OK, I'll do a deeper analysis. I also noticed that the static analyzer gave me a red mark, which I'm going to consider a big red flag.